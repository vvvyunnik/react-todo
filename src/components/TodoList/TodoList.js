import { useSelector } from "react-redux";
import TodoItem from "../TodoItem/TodoItem";
import { selectFilteredTodos } from "../../store/selectors";
import "./TodoList.scss";

function TodoList() {
  const filteredTodos = useSelector(selectFilteredTodos);
  const renderTodos = () => {
    if (filteredTodos?.length === 0) {
      return <p className="no-item-text">No todos</p>;
    }

    return (
      <ul className="todo-list">
        {filteredTodos?.map((todo) => (
          <TodoItem
            key={todo.id}
            id={todo.id}
            text={todo.text}
            completed={todo.completed}
          />
        ))}
      </ul>
    );
  };

  return <div className="todos">{renderTodos()}</div>;
}

export default TodoList;
