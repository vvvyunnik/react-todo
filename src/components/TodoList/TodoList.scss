@use '../../styles/variables.scss' as *;

.todos {
    min-height: 300px;
    min-width: 200px;
    background-color: #ffffff;
    border-radius: 8px;

    .todo-list {
        list-style: none;
        padding: 5px 10px;
    }

    .no-item-text {
        text-align: center;
        padding-top: 50px;
        font-weight: $bold-font;
    }
}

