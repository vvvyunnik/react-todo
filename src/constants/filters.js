const FilterType = Object.freeze({
  None: "None",
  Completed: "Completed",
  Active: "Active",
});

const FilterLabelMap = Object.freeze({
  [FilterType.None]: "ALL",
  [FilterType.Completed]: "COMPLETED",
  [FilterType.Active]: "ACTIVE",
});

const TodoFilters = [FilterType.None, FilterType.Completed, FilterType.Active];

export { FilterType, FilterLabelMap, TodoFilters };
