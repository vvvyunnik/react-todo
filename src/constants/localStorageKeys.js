const LocalStorageKeys = Object.freeze({
  Todo: "todo-items",
});

export default LocalStorageKeys;
