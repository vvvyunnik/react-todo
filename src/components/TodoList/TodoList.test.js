import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import TodoList from "./TodoList";
import { FilterType } from "../../constants/filters";

const mockStore = configureStore([]);

const renderTodoListWithState = (initialState) => {
  const store = mockStore(initialState);
  return render(
    <Provider store={store}>
      <TodoList />
    </Provider>
  );
};

describe("TodoList", () => {
  test("matches snapshot", () => {
    const initialState = {
      todo: {
        items: [
          { id: 1, text: "Task 1", completed: false },
          { id: 2, text: "Task 2", completed: true },
          { id: 3, text: "Task 3", completed: false },
        ],
      },
      filter: { filter: FilterType.None },
    };

    const todoListElement = renderTodoListWithState(initialState);

    expect(todoListElement).toMatchSnapshot();
  });

  test("renders 'No todos' when todos is empty", () => {
    const initialState = {
      todo: {
        items: [],
      },
      filter: { filter: FilterType.None },
    };

    const { getByText } = renderTodoListWithState(initialState);
    const noItemTextElement = getByText("No todos");

    expect(noItemTextElement).toBeInTheDocument();
  });

  test("renders todos", () => {
    const todos = [
      { id: 1, text: "Todo 1", completed: false },
      { id: 2, text: "Todo 2", completed: true },
    ];

    const initialState = {
      todo: {
        items: todos,
      },
      filter: { filter: FilterType.None },
    };

    const { getByText } = renderTodoListWithState(initialState);

    todos.forEach((todo) => {
      const todoItemElement = getByText(todo.text);
      expect(todoItemElement).toBeInTheDocument();
    });
  });
});
