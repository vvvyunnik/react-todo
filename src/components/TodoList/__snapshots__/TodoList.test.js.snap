// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoList matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="todos"
      >
        <ul
          class="todo-list"
        >
          <li
            class="todo-item "
          >
            <input
              class="todo-item-checkbox"
              type="checkbox"
            />
            <span
              class="todo-item-text"
            >
              Task 1
            </span>
            <button
              class="no-style-btn delete-btn"
              type="button"
            >
              <svg>
                cross.svg
              </svg>
            </button>
          </li>
          <li
            class="todo-item completed"
          >
            <input
              checked=""
              class="todo-item-checkbox"
              type="checkbox"
            />
            <span
              class="todo-item-text"
            >
              Task 2
            </span>
            <button
              class="no-style-btn delete-btn"
              type="button"
            >
              <svg>
                cross.svg
              </svg>
            </button>
          </li>
          <li
            class="todo-item "
          >
            <input
              class="todo-item-checkbox"
              type="checkbox"
            />
            <span
              class="todo-item-text"
            >
              Task 3
            </span>
            <button
              class="no-style-btn delete-btn"
              type="button"
            >
              <svg>
                cross.svg
              </svg>
            </button>
          </li>
        </ul>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="todos"
    >
      <ul
        class="todo-list"
      >
        <li
          class="todo-item "
        >
          <input
            class="todo-item-checkbox"
            type="checkbox"
          />
          <span
            class="todo-item-text"
          >
            Task 1
          </span>
          <button
            class="no-style-btn delete-btn"
            type="button"
          >
            <svg>
              cross.svg
            </svg>
          </button>
        </li>
        <li
          class="todo-item completed"
        >
          <input
            checked=""
            class="todo-item-checkbox"
            type="checkbox"
          />
          <span
            class="todo-item-text"
          >
            Task 2
          </span>
          <button
            class="no-style-btn delete-btn"
            type="button"
          >
            <svg>
              cross.svg
            </svg>
          </button>
        </li>
        <li
          class="todo-item "
        >
          <input
            class="todo-item-checkbox"
            type="checkbox"
          />
          <span
            class="todo-item-text"
          >
            Task 3
          </span>
          <button
            class="no-style-btn delete-btn"
            type="button"
          >
            <svg>
              cross.svg
            </svg>
          </button>
        </li>
      </ul>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
